samples/System.Numerics.Tensors.Samples/TensorSamples/PreAllocate.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace TensorSamples
+{
+    partial class PreAllocate
+    {
+        public static void RunSample()
+        {
+            Console.WriteLine($"*** {nameof(PreAllocate)} ***");
+            var multTable = GetMultiplicationTable(5);
+
+            Console.WriteLine("Multiplication table:");
+            Console.WriteLine(multTable.GetArrayString());
+
+            Console.WriteLine("Sums:");
+            for (int row = 0; row < multTable.Dimensions[0]; row++)
+            {
+                Console.WriteLine(GetRowSum(multTable, row));
+            }
+        }
+        
+        // The following represent what .NET bindings would look like for a native library that wishes
+        // to deal in tensors where the size of the tensor is known before calling the API and can be allocated
+        // by the caller.
+        
+        public static unsafe DenseTensor<double> GetMultiplicationTable(int maxNumber)
+        {
+            Span<int> dimensions = stackalloc int[2];
+            dimensions[0] = dimensions[1] = maxNumber;
+
+            var result = new DenseTensor<double>(dimensions);
+            
+            fixed (double* dataPtr = &result.Buffer.Span.DangerousGetPinnableReference())
+            {
+                GetMultTablePreAllocated(maxNumber, dataPtr, result.Buffer.Length);
+            }
+
+            return result;
+        }
+
+        public static unsafe double GetRowSum(DenseTensor<double> tensor, int row)
+        {
+            fixed(double* dataPtr = &tensor.Buffer.Span.DangerousGetPinnableReference())
+            {
+                return GetRowSum(dataPtr, tensor.Dimensions.ToArray(), tensor.Rank, row);
```

Comment:
Why `tensor.Dimensions.ToArray()` and not fix the `Dimensions` span? No need to alloc. `GetRowSum` should be changed accordingly.